sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionBase.scala

```diff
@@ -81,6 +87,66 @@ abstract class TypeCoercionBase extends TypeCoercionHelper {
     }
   }
 
+  object DefaultValueExpressionCoercion extends Rule[LogicalPlan] {
```

Comment:
Could you add a comment for this new rule?

Reply:
added